### DISCUSSION TITLE
Question about CURLMOPT_SOCKETFUNCTION

### DISCUSSION THREAD

**Discussion by traviswsmith** [curl/curl#13300] [Q&A] [answered]
When processing the `what` parameter in a `CURLMOPT_SOCKETFUNCTION` callback, does a `CURL_POLL_OUT` imply **not** `CURL_POLL_IN` and vice-versa?

In other words, if a previous call asked for `CURL_POLL_IN` and a new one asks for `CURL_POLL_OUT` should we stop waiting for incoming data, or wait for both?

Is it guaranteed that `CURL_POLL_INOUT` will be provided in all cases we need to continue listening for a previously registered `what`?

**Comment by bagder** [comment, marked as answer]
The callback gets told what activity to wait for on that specific socket. Any previous activity is replaced by the new call.
